core-rust/state-manager/src/store/rocks_db.rs

```diff
@@ -315,13 +318,21 @@ impl CommitStore for RocksDBStore {
             batch.put_cf(self.cf_handle(&VertexStore), [], vertex_store);
         }
 
-        for (key, node) in commit_bundle.hash_tree_nodes {
+        let state_hash_tree_update = commit_bundle.state_hash_tree_update;
+        for (key, node) in state_hash_tree_update.new_nodes {
             batch.put_cf(
-                self.cf_handle(&HashTreeNodes),
+                self.cf_handle(&StateHashTreeNodes),
                 encode_key(&key),
                 scrypto_encode(&node).unwrap(),
             );
         }
+        for stale_node_keys in state_hash_tree_update.stale_node_keys_at_state_version {
+            batch.put_cf(
+                self.cf_handle(&StaleStateHashTreeNodeKeysByStateVersion),
+                stale_node_keys.0.to_be_bytes(),
+                encode_vec(&stale_node_keys.1, encode_key),
```

Comment:
Note from author:

I chose to store it like this:
```
db_key = state version | db_value = stale node keys
-------------------------------------------------------------------------
                     7 | vec![{v=2; nibbles=a1b}, {v=6; nibbles=a2}]
                     8 | vec![{v=7; nibbles=a1b}]
```
(because that's the way the pruner will most likely want to consume it: "get stale node keys for versions older than current_version-X")

I wonder if there are any downsides of that... because the OTHER open-sourced JMT-using project stores it like this:
```
db_key = node key  | db_value = state version
------------------------------------------------------------
{v=2; nibbles=a1b} | 7
{v=6; nibbles=a2}  | 7
{v=7; nibbles=a1b} | 8
```
(which doesn't have any obvious advantage to me)

Reply:
seems reasonable to store it by state version :+1:

Reply:
seems reasonable to store it by the state version :+1: